File: unload-manager/src/main/java/com/starship/unloadmanager/exception/ExceptionHandlerAdvice.java
package com.starship.unloadmanager.exception;

import com.starship.unloadmanager.dto.BadParamsResponseDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class ExceptionHandlerAdvice {

    @ExceptionHandler(BadRequestException.class)
    public ResponseEntity<BadParamsResponseDto> handleException(BadRequestException e) {
        return ResponseEntity.status(e.getHttpStatus()).body(e.getBadParamsResponseDto());
    }

    @ExceptionHandler(ResourceException.class)
    public ResponseEntity<String> handleException(ResourceException e) {
        return ResponseEntity.status(e.getHttpStatus()).body(e.getMessage());
    }
}

File: unload-manager/src/main/kotlin/com/starship/unloadmanager/exception/ExceptionHandlerAdvice.kt
package com.starship.unloadmanager.exception

import com.starship.unloadmanager.dto.BadParamsResponseDto
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class ExceptionHandlerAdvice {
    @ExceptionHandler(BadRequestException::class)
    fun handleException(e: BadRequestException): ResponseEntity<BadParamsResponseDto> {
        return ResponseEntity.status(e.httpStatus).body(e.badParamsResponseDto)
    }

    @ExceptionHandler(ResourceException::class)
    fun handleException(e: ResourceException): ResponseEntity<String> {
        return ResponseEntity.status(e.httpStatus).body(e.message)
    }
}
